File: backend/src/main/java/fish/app/fishecommerce/model/entity/Fish.java
package fish.app.fishecommerce.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;

@Data
@Entity
public class Fish {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private Integer price;
    private Double weightRangeFrom;
    private Double weightRangeTo;
    private Double amount;
    private String location;

    @Column(columnDefinition = "TEXT")
    private String description;

}

File: frontend/src/model/fish.ts
import { z } from 'zod';


export class Fish {
    id: number;
    name: string;
    price: number;
    weightRangeFrom: number;
    weightRangeTo: number;
    amount: number;
    location: string;
    description: string;

    constructor(
        id: number,
        name: string,
        price: number,
        weightRangeFrom: number,
        weightRangeTo: number,
        amount: number,
        location: string,
        description: string
    ) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.weightRangeFrom = weightRangeFrom;
        this.weightRangeTo = weightRangeTo;
        this.amount = amount;
        this.location = location;
        this.description = description;
    }
};

export const FishDto = z.object({
    id: z.coerce.number(),
    name: z.string(),
    price: z.coerce.number(),
    weightRangeFrom: z.coerce.number(),
    weightRangeTo: z.coerce.number(),
    amount: z.coerce.number(),
    location: z.string(),
    description: z.string(),
});

export const CreateFishDto = FishDto.omit({ id: true });

export const UpdateFishDto = FishDto

